src/nft-price-checker.go

```diff
@@ -1,138 +1,181 @@
 package main
 
 import (
-	"bufio"
 	"encoding/json"
 	"fmt"
 	"io/ioutil"
 	"log"
 	"net/http"
 	"os"
-	"strconv"
 )
 
-type Collection struct {
+//API request storage structs
+type OwnerCollection struct {
+	Slug            string  `json:"slug"`
+	OwnedAssetCount float64 `json:"owned_asset_count"`
+}
+
+type CollectionFloorPrice struct {
 	Stats struct {
 		Floor_price float64 `json:"floor_price"`
 	}
 }
 
-type Price struct {
+type EthPrice struct {
 	Ethereum struct {
 		Usd float64 `json:"usd"`
 	}
 }
 
-type AccountCollection struct {
-	Slug            string  `json:"slug"`
-	OwnedAssetCount float64 `json:"owned_asset_count"`
+//Input file storage structs
+type Collection struct {
+	CollectionID string  `json:"collectionId"`
+	Count        float64 `json:"count"`
+}
+
+type Input struct {
+	OwnerAddress string       `json:"ownerAddress"`
+	Collections  []Collection `json:"collections"`
 }
 
-// Reading input.txt file
-func readInput(path string) ([]string, []string, error) {
-	file, err := os.Open(path)
+func get(url string) []byte {
+	req, err := http.NewRequest("GET", url, nil)
 	if err != nil {
-		return nil, nil, err
+		log.Fatalln(err)
 	}
-	defer file.Close()
+	req.Header.Add("Accept", "application/json")
 
-	var entries []string
-	var nb []string
-	line := 1
-	scanner := bufio.NewScanner(file)
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		log.Fatalln(err)
+	}
 
-	for scanner.Scan() {
-		if line == 1 {
-			if scanner.Text() != "Collections:" {
-				return getCollectionsByWalletAdress(scanner.Text(), scanner.Err())
-			}
-		} else if line != 1 {
-			entries = append(entries, scanner.Text()[:len(scanner.Text())-2])
-			nb = append(nb, scanner.Text()[len(scanner.Text())-1:])
+	defer func() {
+		err := res.Body.Close()
+		if err != nil {
+			log.Fatalln(err)
 		}
-		line++
+	}()
+
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		log.Fatalln(err)
 	}
 
-	return entries, nb, scanner.Err()
+	return body
 }
 
-//Asking API for an account (address)
-func getCollectionsByWalletAdress(adr string, err error) ([]string, []string, error) {
-	var entries []string
-	var nb []string
-	var cols []AccountCollection
+func readInput(path string) ([]Collection, error) {
+	file, err := ioutil.ReadFile(path)
+	if err != nil {
+		return []Collection{}, err
+	}
 
-	url := fmt.Sprintf("https://api.opensea.io/api/v1/collections?asset_owner=%s&offset=0&limit=300", adr)
-	req, _ := http.NewRequest("GET", url, nil)
-	req.Header.Add("Accept", "application/json")
-	res, _ := http.DefaultClient.Do(req)
-	defer res.Body.Close()
-	body, _ := ioutil.ReadAll(res.Body)
+	var inputFile Input
```

Comment:
It looks more like json to me. It is not really the file.
```suggestion
	var jsonInput Input
```